File: plotter.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"strings"

	"golang.org/x/term"
)

func main() {
	args := os.Args[1:]
	if len(args) == 0 {
		fmt.Println("Usage: go run main.go 10 20 30 40")
		return
	}

	// 1. Determine Terminal Size
	width, height, err := term.GetSize(int(os.Stdin.Fd()))
	if err != nil {
		width, height = 80, 24
	}
	
	plotHeight := height - 5
	if plotHeight < 5 {
		plotHeight = 5
	}

	// 2. Construct Gnuplot Commands
	dataString := strings.Join(args, "\n")
	gnuplotCmds := fmt.Sprintf(`
		set terminal dumb size %d %d
		set ylabel "Mb/s"
		set xlabel "Minutes"
		set grid
		plot "-" using 0:1 with lines title "Network Performance"
		%s
		e
	`, width, plotHeight, dataString)

	// 3. Print Green ANSI Escape Code
	fmt.Print("\033[32m") 

	// 4. Execute Gnuplot
	cmd := exec.Command("gnuplot")
	cmd.Stdin = strings.NewReader(gnuplotCmds)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	if err := cmd.Run(); err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
	}

	// 5. Reset Terminal Color
	fmt.Print("\033[0m")
}

File: plotter.py
#!/usr/bin/env python3
import click
import subprocess
import shutil

@click.command()
@click.argument('numbers', nargs=-1, type=float)
def plot_data(numbers):
    """Detects terminal size and plots numbers with custom labels."""
    if not numbers:
        click.echo("Usage: python script.py 10 20 30 40")
        return

    # 1. Determine Terminal Size
    size = shutil.get_terminal_size()
    width = size.columns
    height = max(5, size.lines - 4)  # Leave room for labels/prompt

    # 2. Prepare Data
    data_string = "\n".join(str(n) for n in numbers)

    # 3. Gnuplot Commands
    # 'set xlabel' and 'set ylabel' add the axis descriptions
    gnuplot_commands = [
        f'set terminal dumb size {width} {height}',
        'set ylabel "Mb/s"',
        'set xlabel "Minutes"',
        'plot "-" using 0:1 with lines title "Network Performance"',
        'set grid',
        data_string,
        'e'
    ]

    command_input = "\n".join(gnuplot_commands)

    try:
        subprocess.run(['gnuplot'], input=command_input, text=True, check=True)
    except FileNotFoundError:
        click.echo("Error: Gnuplot not found in PATH.", err=True)

if __name__ == '__main__':
    plot_data()

